test/jdk/com/sun/net/httpserver/SelCacheTest.java

```diff
@@ -36,36 +38,46 @@
 import jdk.test.lib.net.SimpleSSLContext;
 import jdk.test.lib.net.URIBuilder;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.concurrent.*;
 import java.io.*;
 import java.net.*;
 import javax.net.ssl.*;
 
+import static jdk.test.lib.Utils.createTempFileOfSize;
+
 /* basic http/s connectivity test
  * (based on Test1)
  */
 
 public class SelCacheTest extends Test {
 
+    private static final String TEMP_FILE_PREFIX =
+            HttpServer.class.getPackageName() + '-' + SelCacheTest.class.getSimpleName() + '-';
+
     static SSLContext ctx;
 
     public static void main(String[] args) throws Exception {
         HttpServer s1 = null;
         HttpsServer s2 = null;
         ExecutorService executor=null;
+        Path smallFilePath = createTempFileOfSize(TEMP_FILE_PREFIX, null, 23);
+        Path largeFilePath = createTempFileOfSize(TEMP_FILE_PREFIX, null, 2730088);
         InetAddress loopback = InetAddress.getLoopbackAddress();
         try {
-            String root = System.getProperty("test.src")+ "/docs";
             System.out.print("Test1: ");
             InetSocketAddress addr = new InetSocketAddress(loopback, 0);
             s1 = HttpServer.create(addr, 0);
             if (s1 instanceof HttpsServer) {
                 throw new RuntimeException("should not be httpsserver");
             }
             s2 = HttpsServer.create(addr, 0);
-            HttpHandler h = new FileServerHandler(root);
-            HttpContext c1 = s1.createContext("/test1", h);
-            HttpContext c2 = s2.createContext("/test1", h);
+            // Assert that both files share the same parent and can be served from the same `FileServerHandler`
+            assert smallFilePath.getParent().equals(largeFilePath.getParent());
```

Comment:
I see what you are doing here. On the one hand I thought that it would be better placed right after line 66, but on the other hand we want it inside the `try { }` so that files will be deleted on `finally` even if it fires. So LGTM.